packages/contrast-colors/index.js

```diff
@@ -13,7 +13,480 @@ governing permissions and limitations under the License.
 const d3 = require('./d3.js');
 
 const { catmullRom2bezier, prepareCurve } = require('./curve.js');
-const { color } = require('./d3.js');
+// const { color } = require('./d3.js');
+
+class Theme {
+  constructor({colors, backgroundColor, lightness, contrast = 1, output = 'HEX'}) {
+    this._output = output;
+    this._colors = colors;
+    this._lightness = lightness;
+
+    this._setBackgroundColor(backgroundColor);
+    this._setBackgroundColorValue();
+    
+    this._contrast = contrast;
+    if (!this._colors) {
+      throw new Error(`No colors are defined`);
+    }
+    if (!this._backgroundColor) {
+      throw new Error(`Background color is undefined`);
+    }
+    colors.forEach(color => {
+      if(!color.ratios) throw new Error(`Color ${color.name}'s ratios are undefined`);
+    });
+    if (!colorSpaces[this._output]) {
+      throw new Error(`Output “${colorspace}” not supported`);
+    }
+
+    this._modifiedColors = this._colors;
+    // console.log(`${this._colors} \n ----------------- \n ${this._modifiedColors}`)
+    // this._setContrasts(this._contrast);
+    
+    this._findContrastColors();
+    this._findContrastColorValues();
```

Comment:
Are these the heavy functions that take CPU time? I wonder if you want to put off that work until the first getting or function is called so that creating an instance of `Theme` is a fast operation. You would likely need to implement a `_initialCalculations()` function that checks a boolean `this._initialCalculationsPerformed` and calls these functions if not.

It all depends on when you want the work done.